lib/utils/http.go

```diff
@@ -176,3 +178,52 @@ func (m *maxBytesReader) Read(p []byte) (int, error) {
 	}
 	return n, err
 }
+
+// Sensitive headers list based on team discussion and Akami's list of
+// not-to-be-logged headers here:
+//    https://techdocs.akamai.com/edge-diagnostics/reference/sensitive-headers
+//
+//  - Authorization
+//  - Proxy-Authorization
+//  - Set-Cookie
+//  - Anything containing "API-Key"
+//  - X-Amz-Security-Token
+
+// sensitiveHeaderKeys is the list HTTP headers deemed to be too sensitive
+// to be written to a log.
+var sensitiveHeaderKeys []string = []string{
+	"authorization",
+	"proxy-authorization",
+	"set-cookie",
+	"x-amz-security-token",
+}
+
+// sensitiveHeaderFragments is a list of suspect header fragments. If a header
+// key contains any of these fragments it will be filtered out by
+// SanitizeHeaders()
+var sensitiveHeaderFragments []string = []string{
+	"api-key",
+}
+
+// SanitizeHeaders filters sensitive out of a an HTT message's header collection,
+// in order to make them suitable for writing to logs. Returns a new http.Header
+// containing the filtered collection.
+func SanitizeHeaders(src http.Header) http.Header {
```

Comment:
Probably fine, but this approach seems a bit heavy. We're allocating a new map and making an allocation to copy each non-sensitive header, just to log it an then throw it away.

Have you thought about implementing a `slog.LogValuer` instead? This would be more fit for purpose - it's used only for logging and lazily evaluated so no unnecessary work needs to be done if the log doesn't need to happen.

Reply:
Had definitely thought of that, but as the log site was still using `logrus` I didn't pursue it. Probably a better approach going forward, though - and i can always use it explicitly at the existing call site.

Reply:
Re: allocating a new map, etc - yeah, it's a bit heavyweight but formatting the headers directly into a string posed a bunch of interesting testing problems thanks to Go's arbitrary map traversal order. I figured that sorting the Header keys to make it easier to test was even more heavyweight than allocating the new map.

Reply:
Anyway, I've added a wrapper around `http.Header` that we can use to defer the header sanitising until it's actually needed.